content/automate/ManageIQ/Cloud/VM/Retirement/Email.class/__methods__/vm_retire_extend.rb

```diff
@@ -3,63 +3,82 @@
-
-if vm.retires_on.nil? | vm.retires_on == ""
-  $evm.log("info", "VM '#{vm_name}' has no retirement date - extension bypassed")
-  exit MIQ_OK
-end
-
-if vm.retired
-  $evm.log("info", "VM '#{vm_name}' already marked as retired. vm.retires_on date: #{vm.retires_on}. No Action taken")
-  exit MIQ_OK
-end
-
-$evm.log("info", "VM: <#{vm_name}> current retirement date is #{vm.retires_on}")
-
-unless vm.retires_on.nil?
-  $evm.log("info", "Extending retirement <#{vm_retire_extend_days}> days for VM: <#{vm_name}>")
-
-  # Set new retirement date here
-  vm.extend_retires_on(vm_retire_extend_days, vm.retires_on)
-
-  $evm.log("info", "VM: <#{vm_name}> new retirement date is #{vm.retires_on}")
-
-  $evm.log("info", "Inspecting retirement: <#{vm.retirement.inspect}>")
-
-  evm_owner_id = vm.attributes['evm_owner_id']
-  owner = nil
-  owner = $evm.vmdb('user', evm_owner_id) unless evm_owner_id.nil?
-  $evm.log("info", "Inspecting VM Owner: #{owner.inspect}")
-
-  if owner
-    to = owner.email
-  else
-    to = $evm.object['to_email_address']
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Retirement
+          module Email
+            class VmRetireExtend
+              def initialize(handle = @evm)
+                @handle = handle
+              end
+
+              def main
+                @handle.log("info", "Starting vm_retire_extend")
+                vm = @handle.root['vm']
+                check_retire_extend(vm)
+                @handle.log("info", "Ending vm_retire_extend")
+              end
+
+              private
+
+              def check_retire_extend(vm)
+                vm_retire_extend_days = @handle.object['vm_retire_extend_days']
+                raise "ERROR - vm_retire_extend_days not found!" if vm_retire_extend_days.nil?
+
+                @handle.log("info", "Number of days to extend: <#{vm_retire_extend_days}>")
+
+                vm_name = vm.name
+
+                if vm.retires_on.blank?
+                  @handle.log("info", "VM '#{vm_name}' has no retirement date - extension bypassed")
+                  exit MIQ_OK
+                end
+
+                if vm.retired
+                  @handle.log("info", "VM '#{vm_name}' is already retired. retires_on date: #{vm.retires_on}. No Action taken")
+                  exit MIQ_OK
+                end
+
+                @handle.log("info", "VM: <#{vm_name}> current retirement date is #{vm.retires_on}")
+                @handle.log("info", "Extending retirement <#{vm_retire_extend_days}> days for VM: <#{vm_name}>")
+
+                vm.extend_retires_on(vm_retire_extend_days, vm.retires_on)
+
+                @handle.log("info", "VM: <#{vm_name}> new retirement date is #{vm.retires_on}")
+                @handle.log("info", "Inspecting retirement vm: <#{vm.retirement_state.try}>")
+
+                evm_owner_id = vm.attributes['evm_owner_id']
+                owner = @handle.vmdb('user', evm_owner_id) unless evm_owner_id.nil?
```

Comment:
@billfitzgerald0120 
Should this be
owner = @handle.vmdb('user', evm_owner_id) if evm_owner_id

Otherwise its double negation with the unless and nil